news/2025/12/05/ModelingToolkitv11.md

```diff
@@ -0,0 +1,272 @@
+of AGPL are satisfied. It was already the case before that some dependencies of ModelingToolkit.jl and
+Julia itself were GPL dependencies, notably SuiteSparse and FFTW are GPL-licensed libraries and thus
+ModelingToolkit.jl already had some GPL implications. However, with this change, the implications include
+the symbolic transformation libraries specifically for handling acausal models and high-index DAEs.
+
+This was a difficult decision to make, and the evolution of ModelingToolkit.jl shows many scars of previous
+failed attempts to do this correctly. We previously had StructuralTransformation.jl as a separate package
+back in the early days of ModelingToolkit.jl, but ultimately the developer burden of maintaining it separately
+was so high that it had almost killed the entire project. Because of this, we had always had to hold back
+on actively releasing some of the more advanced features into the ecosystem. Some of the coolest features
+and major advancements in acausal modeling, such as 
+[techniques for making large loops and arrays compile fast](https://patents.google.com/patent/US20250013444A1/en),
+specifically had to be omitted or kept separate in order to deal with licensing issues due to the commerical
+aspects for which they were developed. Many may recall the project JuliaSimCompiler.jl, which was a separate
+backend to ModelingToolkit.jl models that implemented many of these compiler passes. Ultimately, having to
+maintain two separate compilers means having 2 bad compilers: during the time frame of JuliaSimCompiler the
+issues list in both repositories grew enormously and the bug surface became very high due to divergences in
+compiler support. Once more, having this separate compiler almost killed the entire project due to the high
+maintanance burden.
+
+Thus we had to spend some time to really think through how to do this effectively in a way that would sustainable
+for the future of the project. Ultimately the issue is that, unlike what has occured in the solver ecosystem, 
+almost all of the funding that has been achieved for ModelingToolkit.jl throughout its entire lifespan has had
+both developer and government requirements for commercialization. While many may believe that the (US) government
+supports open source software development and would prefer for this software to be distributed as free under a
+permissive license, the reality is that government contracts and grants often have requirements for 
+commercialization. For full transparency, ModelingToolkit.jl has required about 3 full-time expert/senior
+developers for the last 5 years, and the funding for this project has totaled around ~$250k (after overhead) of
+government grants directed to the non-commercial aspects of the project (we thank Bernt Lie for his early
+contributions via DigiWell and Chris Tessum for co-PIing on a related NSF grant). We additionally note that 
+total donations to the project have been <$10k over the full 10 years. 
```

Comment:
This is not being fully honest about the situation *in my opinion*. The post on Discourse, and the comments when it was reviewed by JuliaHub, made clear that the primary motivator here is concerns about the behavior of JuliaHub competitors (as we said in that post). The grant funding issue was not something that anyone else from JuliaHub commented on (and even significantly cut down from the original draft Discourse post). While I agree that having grant funding to carry out this work outside of JuliaHub would make MTK sustainable on its own, it was not the immediate reason for this change (or even the first reason presented to the SciML council for why this change was necessary).

Reply:
I would say it is "all of the above" - but I do agree that this is better off not being in here.